hastory-cli/bench/Main.hs

```diff
@@ -46,8 +46,7 @@ main =
         ]
 
 benchSets :: Settings
-benchSets =
-  Settings {setCacheDir = $(mkAbsDir "/tmp/hastory-cache"), remoteStorageClientInfo = Nothing}
+benchSets = Settings {setCacheDir = $(mkAbsDir "/tmp/hastory-cache")}
```

Comment:
@NorfairKing I may be missing things completely here, but my current understanding is that [withSystemTempDir](https://hackage.haskell.org/package/path-io-1.6.0/docs/Path-IO.html#v:withSystemTempDir) is the wrong function to use because it will clean up after calling the function that uses the resource / temp dir. We would return a `Settings` object where the cache dir is set to a directory that no longer exists due to the aforementioned cleanup. 

If I am correct there, then I think the best way forward is to replace the use of `env` with [envWithCleanup](https://hackage.haskell.org/package/criterion-1.5.7.0/docs/Criterion.html#v:envWithCleanup). In so doing, we'd manually create a hastory-specific temp dir using [createTempDir](https://hackage.haskell.org/package/path-io-1.6.0/docs/Path-IO.html#v:createTempDir) and clean up after ourselves.

Reply:
New issue #38 has been created.